action.yml

```diff
@@ -1,60 +1,188 @@
+  openedx_extra_pip_requirements:
+    description: "Optional extra pip requirements to install in Open edX. E.g: 'package1==1.0 package2>=2.0'"
+    required: false
+    default: ""
+  fixtures_file:
+    description: "Optional path to the plugin's fixtures file to load."
+    required: false
+  openedx_imports_test_file_path:
+    description: "Path to the file that contains the test function for validating Open edX imports. This should be a Python file within your project."
+    required: false
+  openedx_imports_test_function_name:
+    description: "Name of the function in the specified file that executes the import tests for Open edX."
     required: false
 
 runs:
-  using: 'composite'
+  using: "composite"
   steps:
-  - name: Create a Virtual Environment
-    run: |
-      pip install virtualenv
-      virtualenv venv
-    shell: bash
-
-  - name: Install Tutor
-    run: |
-      source venv/bin/activate
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        git clone --branch=nightly --depth=1 https://github.com/overhangio/tutor.git
-        chmod 777 tutor -R
-        pip install -e "./tutor[full]"
-      else
-        pip install "tutor$INPUT_TUTOR_VERSION"
-      fi
-    shell: bash
-    env:
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Configure Tutor and Launch
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor --version
-      TUTOR_ROOT="$(pwd)" tutor config save
-      TUTOR_ROOT="$(pwd)" tutor mounts add lms,cms,lms-worker,cms-worker:$(pwd)/$INPUT_APP:/openedx/$INPUT_APP
-      chmod 777 . -R
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        TUTOR_ROOT="$(pwd)" tutor images build openedx
-      fi
-      TUTOR_ROOT="$(pwd)" tutor local launch -I
-    shell: bash
-    env:
-      INPUT_APP: ${{ inputs.app_name }}
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Run integration tests in lms
-    if: ${{ inputs.shell_file_to_run }}
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor local run lms bash $INPUT_SHELL_FILE
-    shell: bash
-    env:
-      INPUT_SHELL_FILE: /openedx/${{ inputs.app_name }}/${{ inputs.shell_file_to_run }}
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        path: ${{ inputs.app_name }}
+
+    - name: Adjust permissions to execute Tutor commands
+      run: |
+        chmod 777 . -R
+      shell: bash
+
+    - name: Set Tutor environment variables
+      run: |
+        echo "LMS_HOST=local.edly.io" >> "$GITHUB_ENV"
+        echo "CMS_HOST=studio.local.edly.io" >> "$GITHUB_ENV"
+        echo "TUTOR_ROOT=$(pwd)" >> "$GITHUB_ENV"
+        echo "TUTOR_PLUGINS_ROOT=$(pwd)/plugins/" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Create virtualenvs
+      run: |
+        echo "Creating isolated venv for Tutor"
+        python -m venv .tutor_venv
+        echo "Creating isolated venv to run the integration tests"
+        python -m venv .tests_venv
```

Comment:
Why do we need to use two environments?

Reply:
tutor_venv: This env is specifically for running Tutor to avoid permission problems. Without this, installing Tutor (especially the nightly version as an editable package) would try to install in the system-wide Python path, which causes problems with the permissions in the runner

tests_venv: This is used for executing the integration tests in a fully isolated environment. We do this because by separating the test environment, we prevent any potential conflicts with the Tutor environment and ensure that the tests run with only  the necessary dependencies

Reply:
So, using tutor_env fixes the issue, which is also solved by adjusting permissions [here](https://github.com/eduNEXT/integration-test-in-tutor/pull/4/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R36). Or are those unrelated?

Reply:
@mariajgrimaldi  They are somewhat related but not interchangeable. Based on my testing while refactoring, here’s what I found:

The reason for using a venv for Tutor is to avoid permission issues when installing it as an editable package. Without the venv, `pip install -e` tries to install Tutor in the system-wide Python path, which requires root access. Changing permissions with `chmod` doesn’t affect system-wide directories (unless we do something like `sudo chmod 777 /usr/local/ -R`, which is not recommended), so that's why the venv is necessary in this case.

On the other hand, `chmod 777` is needed when installing the plugin as an editable package inside the Tutor containers. When running `pip install -e`, Python tries to create files like `*.egg-info` but doesn't have sufficient permissions to write to those directories inside the container

I can try narrowing the `chmod` to specific directories instead of applying it globally, which might help reduce the scope of the permission changes.

Here’s a link to the output when I only used `chmod` without the isolated env for Tutor installation:  
https://github.com/eduNEXT/eox-core/actions/runs/11133124056/job/30938507427?pr=283

And here’s the output when using the venv but without `chmod`:  
https://github.com/eduNEXT/eox-core/actions/runs/11133257345/job/30938936846?pr=283

I tried to get rid of one or the other (especially `chmod`, which I'd prefer to remove), but none of my attempts worked. I’ll open an issue so we can keep investigating and see if there’s a better solution

Reply:
Thank you so much for the thorough investigation, it's really helpful for future reference. For me, it's enough to leave this well documented in an issue, as you suggested.